src/js/pages/Home/index.js

```diff
@@ -58,6 +63,7 @@ export default class Home extends React.PureComponent {
    * Functional component
    */
   render() {
+    const { multipleShabads, setMultipleShabads, clearMultipleShabads, removeMultipleShabads, isHome } = this.props;
```

Comment:
I would suggest reconsidering this logic here, we are connecting the store to Homepage now which may cause re-renders of this page.
Please check if you can use these inside the `MultipleShabadsDisplay` component?

Can discuss it further.

Reply:
Connected Individual Component instead giving whole control to Home Component.
Thanks!!